engine/src/main/java/org/terasology/particles/components/ParticleSystemComponent.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.particles.components;
+
+import org.lwjgl.util.vector.Vector2f;
+import org.terasology.entitySystem.Component;
+import org.terasology.entitySystem.Owns;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.particles.events.ParticleSystemUpdateEvent;
+import org.terasology.particles.rendering.ParticleRenderer;
+import org.terasology.rendering.assets.texture.Texture;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Represents a particle system. A particle system must have emitters in order to produce particles.
+ */
+public class ParticleSystemComponent implements Component {
+
+    /**
+     * Reference to the entity this component is attached to
+     */
+    public EntityRef ownerEntity;
+
+    /**
+     * The renderer that should be used to render this system.
+     */
+    private Class<ParticleRenderer> renderer;
+
+    /**
+     * The lifetime of this system
+     */
+    public float maxLifeTime = Float.POSITIVE_INFINITY;
+
+    /**
+     * Destroy this particle system when its last emitter is destroyed TODO: Unimplemented
+     */
+    public boolean destroyWhenFinished = true;
+
+    /**
+     * The amount of particles in this system's pool. Emitters under this system will share the particles.
+     */
+    public int particlePoolSize = 1000;
+
+    @Owns
+    private List<EntityRef> emitters = new LinkedList<>();
+
+    @Owns
+    private List<EntityRef> affectors = new LinkedList<>();
+
+    /**
+     * This system's particle texture
+     */
+    public Texture texture = null;
+
+    /**
+     * This system's particle texture size, in percents x: [0.0, 1.0], y: [0.0, 1.0]
+     */
+    public Vector2f textureSize = new Vector2f(1.0f, 1.0f);
+
+
+    /**
+     * Notifies ParticleSystemManagerImpl that this system needs an updated ParticleSystemStateData
+     */
+    private void requestUpdate() {
+        if (ownerEntity != null) {
+            ownerEntity.send(new ParticleSystemUpdateEvent());
+        }
+    }
+
+    /**
+     * Changes the particle system's desired renderer. Will register a new renderer if it does not already exist.
+     *
+     * @param renderer The renderer to set this system to.
+     */
+    public void setRenderer(Class<ParticleRenderer> renderer) {
+        this.renderer = renderer;
+        requestUpdate();
```

Comment:
And I'm not sure a ParticleSystemComponent should hold a renderer. For example the same particles might need to be rendered in more than one way - by more than one renderer. 

Rather, a registered RenderSystem should iterate and render all entities having a specific ParticleDataComponent and/or a specific ParticleRenderingComponent attached, the latter providing information shared by all particles that need to be rendered in a certain way, i.e. as the flames of a torch.